libsplinter/src/biome/credentials/store/error.rs

```diff
@@ -36,8 +36,9 @@ pub enum CredentialsStoreError {
     /// Represents general failures in the database
     StorageError {
         context: String,
-        source: Box<dyn Error>,
+        source: Option<Box<dyn Error>>,
```

Comment:
A few typos in the commit message: s/triat/trait, s/benifit/benefit, s/Posion/Poison